### PR TITLE
feat: support geo_values (single geo_type)

### LINKED ISSUE #221: Accept a list of geo_values instead of only a single value
Currently, the covidcast endpoint accepts a single `geo_value` that can either be `*` (all locations) or the identifier of a single location. That means that if an API client wants to fetch, say, the top 500 counties by case counts, they either have to send one query and filter it or send 500 separate queries. The R and Python clients currently do the latter, but it's slow. (Fetching one day of data for 500 counties would take as long as fetching all state data for 500 days -- making it unusably slow if you want more than one or two days of data.)

Can `geo_value` take a comma-separated list, like the issues and time_values parameters?

For new people, the code lives here: https://github.com/cmu-delphi/delphi-epidata/blob/main/src/server/api.php#L940

### PR DESCRIPTION
**Pull request by sgratzl**
closes #221 (single geo type only for now)

add support for `geo_values`.

valid variants:
 * `geo_value=*`
 * `geo_value=CA`
 * `geo_values=CA,NY`

### CODE REVIEW

File: src/server/api.php [hunk 1]
```diff
@@ -959,12 +958,16 @@ function get_covidcast($source, $signals, $time_type, $geo_type, $time_values, $
...
 
-  if ($geo_value === '*') {
+  if ($geo_values === '*') {
     // the wildcard query should return data for all locations in `geo_type`
     $condition_geo_value = 'TRUE';
+  } else if (is_array($geo_values)) {
+    // return data for multiple location
+    $condition_geo_value = filter_strings('t.`geo_value`', $geo_values);
   } else {
     // return data for a particular location
```
Comment (sgsmob) [inline src/server/api.php:968]: If a `'*'` is present in the list will it be interpreted literally (matching `geo_values` that are `'*'`)?

File: integrations/server/test_covidcast.py [hunk 2]
```diff
@@ -311,6 +311,100 @@ def test_location_wildcard(self):
...
+    self.assertEqual(r['message'], 'no results')
+    r = fetch(['11111', '22222'])
+    self.assertEqual(r['message'], 'success')
+    self.assertEqual(r['epidata'], [counties[0], counties[1]])
+    r = fetch(['11111', '33333'])
+    self.assertEqual(r['message'], 'success')
+    self.assertEqual(r['epidata'], [counties[0], counties[2]])
+    r = fetch(['11111', '55555'])
+    self.assertEqual(r['message'], 'success')
+    self.assertEqual(r['epidata'], [counties[0]])
```
Comment (krivard) [inline integrations/server/test_covidcast.py:403]: This is probably the right choice (silently omit bogus geo values from the response) but we should revisit it when we figure out deletion encoding. One of the goals of deletion encoding is to start giving better feedback about "you asked for a bogus county" vs "that information was withheld for privacy reasons", and an error on *parts* of a query might have to be treated carefully.

**Review by krivard — APPROVED** [review]
Everything builds, tests pass, just added note so we don't forget about partial-failures when we fix #195

File: integrations/server/test_covidcast.py [hunk 3]
```diff
@@ -311,6 +311,100 @@ def test_location_wildcard(self):
...
+      'value': 30,
+      'stderr': 31,
+      'sample_size': 32,
+      'direction': 33,
+      'issue': 20200414,
+      'lag': 0,
+      'signal': 'sig',
+    }]
+
+    r = fetch('*')
```
Comment (sgsmob) [inline integrations/server/test_covidcast.py:387]: Please add comments to each call to `fetch()` indicating the specific behavior being tested.

**Review by sgsmob — CHANGES_REQUESTED** [review]

File: integrations/server/test_covidcast.py (hunk 2 above)
Comment (sgratzl) [reply integrations/server/test_covidcast.py:403]: I think it behaves the same way for all kind of parameters that are not existing, e.g., if you specify a wrong data source you just get no results back but it doesn't throw an error. That is the current nature of just generating a where clause

File: src/server/api.php (hunk 1 above)
Comment (sgratzl) [reply src/server/api.php:968]: atm. yes, there is the clear distinction see PR comment about the variants

**Review by sgsmob — APPROVED** [review]

**Pull request merged**